frontend/src/Components/AlgInputs/AlgInputsCBA.js

```diff
@@ -50,141 +46,121 @@ class AlgInputsCBA extends Component {
             profileNames: [],
             profileData: [],
             profileName: "",
-            county: "",
             loadForecastResults: [],
             processedLoadForecastResults: [],
             shouldRender: false
         };
     }
 
     componentDidMount() {
-        axios("http://127.0.0.1:8000/api/config/load_forecast/")
+        axios("http://127.0.0.1:8000/api/config/load_forecast")
             .then(res => {
-                this.setState({ profileData: res.data });
                 const profiles = res.data;
                 const profileNames = [];
-                for (var i = 0; i < res.data.length; i++) {
-                    const profileNamesUnit = {id: "", name: ""};
-                    profileNamesUnit.id = profiles[i]["id"];
-                    profileNamesUnit.name = profiles[i]["config_name"];
-                    profileNames.push(profileNamesUnit);
+                if (profiles.length > 0) {
+                    for (var i = 0; i < profiles.length; i++) {
+                        const profileNamesUnit = {id: "", name: ""};
+                        profileNamesUnit.id = profiles[i]["id"];
+                        profileNamesUnit.name = profiles[i]["config_name"];
+                        profileNames.push(profileNamesUnit);
+                    }
+                    this.setState({ profileData: profiles, profileNames: profileNames, profileName: document.getElementById("standard-profile").value });
```

Comment:
The `profileName` is saved in the state and referenced as TextField with a `onChange` event already set. You should never have to query the DOM. Just get the value from the state.

As the value in the UI changes, the onChange is triggered and the setProfileName is called and should always have the latest value. Remove all the dom queries and replace with the `this.state.profileName`